packages/grid/x-data-grid/src/models/api/gridRowApi.ts

```diff
@@ -24,6 +24,12 @@ export interface GridRowApi {
    * @param {GridRowModel[]} rows The new rows.
    */
   setRows: (rows: GridRowModel[]) => void;
+  /**
+   * Moves a row from its original position to the position given by `targetIndexPosition`.
+   * @param {GridRowId} rowId The row id
+   * @param {number} targetIndexPosition The new position (0-based).
+   */
+  setRowIndex: (rowId: GridRowId, targetIndexPosition: number) => void;
```

Comment:
```suggestion
  setRowIndex: (rowId: GridRowId, targetPosition: number) => void;
```

or

```suggestion
  setRowIndex: (rowId: GridRowId, targetIndex: number) => void;
```

Same for `oldIndexPosition` on the method body.